websites/K/KOLEO/presence.ts

```diff
@@ -0,0 +1,978 @@
+			details: "Ładowanie.",
+			largeImageKey: Assets.Logo,
+			smallImageText: "Ładowanie...",
+			smallImageKey: Assets.Train,
+			startTimestamp: browsingTimestamp,
+		},
+		privacySetting = await presence.getSetting<boolean>("privacy");
+
+	let currentPath;
+
+	if (href.startsWith("https://koleo.pl")) {
+		currentPath = href.replace("https://koleo.pl", "");
+		presenceData.details = "Szuka połączenia rozkładu koleji.";
+		if (
+			href.endsWith("koleo.pl/") ||
+			href.endsWith("koleo.pl/#") ||
+			currentPath.includes("rozklad-jazdy") ||
+			operators.includes(currentPath.split("/")[1])
+		) {
+			let startText: string, endText: string, dateText: string;
+
+			if (href.endsWith("koleo.pl/") || href.endsWith("koleo.pl/#")) {
+				const startStation = document.querySelector<HTMLInputElement>(
+						"#query_start_station"
+					),
+					endStation = document.querySelector(
+						"#query_end_station"
+					) as HTMLInputElement,
+					date = document.querySelector("#query_date") as HTMLInputElement;
+				if (startStation && endStation && date) {
+					startText = startStation.value;
+					endText = endStation.value;
+					dateText = date.value;
+					presenceData.state = "W całym KOLEO...";
+				}
+			} else if (
+				currentPath.includes("rozklad-jazdy") ||
+				operators.includes(currentPath.split("/")[1])
+			) {
+				const startStationButton = document.querySelector(".closest-station"),
+					endStationButton = document.querySelector(".swap-stations"),
+					dateInputWrapper = document.querySelector(
+						".form-date-input__input-wrapper"
+					);
+
+				if (startStationButton && endStationButton && dateInputWrapper) {
+					startText = (
+						startStationButton
+							.closest(".icon-input-action")
+							.querySelector("input") as HTMLInputElement
+					).value;
+					endText = (
+						endStationButton
+							.closest(".icon-input-action")
+							.querySelector("input") as HTMLInputElement
+					).value;
+					dateText = (
+						dateInputWrapper.querySelector("input") as HTMLInputElement
+					).value;
+
+					presenceData.state = `W ${
+						(
+							document.querySelector(".top-banner__heading") || {
+								textContent: "caŁym KOLEO",
+							}
+						).textContent
+					}...`;
+				}
+			}
+			presenceData.smallImageText = "Szuka połączenia...";
+			presenceData.buttons = [
+				{ label: "Rozkład jazdy", url: "https://koleo.pl/rozklad-jazdy" },
+			];
+
+			if (!privacySetting) {
+				if (startText && endText)
+					presenceData.details = `Szuka połączenia z ${startText} do ${endText} na ${dateText}`;
+				else if (startText && !endText)
+					presenceData.details = `Szuka połączenia z ${startText} na ${dateText}`;
+				else if (!startText && endText)
+					presenceData.details = `Szuka połączenia do ${endText} na ${dateText}`;
+			}
+		} else if (currentPath.startsWith("/rozklad-pkp")) {
+			presenceData.state = "W rozkładzie PKP...";
+			const startStation = document.querySelector(
+					"##query_start_station"
```

Comment:
Verify that this selector is accurate. Perhaps it should be the following?
```suggestion
					"#query_start_station"
```